### ISSUE TITLE
Sticky behaviour gets lost after moving to empty tag

### ISSUE THREAD

**Issue by dhaval15** [awesomeWM/awesome#3102]
Output of `awesome --version`:

awesome v4.3-826-g29f6387de (Too long)
 • Compiled against Lua 5.3.5 (running with Lua 5.3)
 • API level: 4
 • D-Bus support: yes
 • xcb-errors support: no
 • execinfo support: yes
 • xcb-randr version: 1.6
 • LGI version: 0.9.2
 • Transparency enabled: yes
 • Custom search paths: no

**How to reproduce the issue:**

I am making client sticky and ontop with shortcut with

```lua
if c.pinned then               
           c.floating = false
           c.sticky = false
           c.ontop = false
           c.pinned = false
else
           c.floating = true  
           c.sticky = true
           c.ontop = true
           c.pinned = true
end
```

**Actual result:**

Works great, but after moving to empty tag, it restore backs and loses its sticky property. 

**Expected result:**

It should not lose stickiness on while moving to empty tags.

**Comment by psychon** [comment]
I added the following to the end of the default config:
```lua
client.connect_signal("manage", function(c)
           c.floating = true
           c.sticky = true
           c.ontop = true
           c.pinned = true
end)
```
Then I started awesome and an xterm. The xterm got these properties applied (I guess, I didn't really check much). Then I switched to other tags via the taglist. The xterm stays visible since it was sticky. All of these tags were empty (well, except for tag `1`).

So: Cannot reproduce with v4.3-809-g5d1394b91
